### PR TITLE
Renders resource even when error. Fix search related reducers so that…

### PR DESCRIPTION
**Pull request by justinlittman**
… trigger renders due to state change.

refs #1730

### CODE REVIEW

**Comment by justinlittman** [comment]
This addresses 2 problems related to #1730:
* After creating a resource from a resource template and returning to the resource template page, resource template search wasn't working.
* User request to not "lose" an entire resource when there was an error from a nested resource template.

**Comment by justinlittman** [comment]
Also, fixed an html id that was set on a table that could have multiple instances (so the id was not unique).

File: __tests__/components/editor/ResourceTemplate.test.js [hunk 1]
```diff
@@ -37,7 +38,17 @@ describe('<ResourceTemplate />', () => {
...
   it('displays Alert', () => {
+    expect(wrapper.find(Alerts).length).toEqual(1)
+  })
+
+  describe('When there is an error and no resource template', () => {
     const alertWrapper = shallow(<ResourceTemplate.WrappedComponent errors={['Oooops']} />)
-    expect(alertWrapper.find(Alerts).length).toEqual(1)
+    it('displays Alert', () => {
+      expect(alertWrapper.find(Alerts).length).toEqual(1)
+    })
```
Comment (mjgiarlo) [inline __tests__/components/editor/ResourceTemplate.test.js]: This looks like the same stuff on lines 17 + 40-42. Intended to test it in two places, or cruft?

Comment (justinlittman) [reply __tests__/components/editor/ResourceTemplate.test.js]: Different tests. In the first case, there is a resource template; in the second there is not. The code handles differently.

**Pull request merged**